astronomer/providers/amazon/aws/hooks/emr.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, Dict, Optional
+
+from botocore.exceptions import ClientError
+
+from astronomer.providers.amazon.aws.hooks.base_aws_async import AwsBaseHookAsync
+
+
+class EmrStepSensorHookAsync(AwsBaseHookAsync):
+    """
+    A thin wrapper to interact with AWS EMR API
+
+    Additional arguments may be specified and are passed down to the underlying AwsBaseHook.
+
+    For more details see here.
+        - airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook
+    """
+
+    def __init__(self, job_flow_id: str, step_id: str, *args: Any, **kwargs: Any) -> None:
+        kwargs["client_type"] = "emr"
+        kwargs["resource_type"] = "emr-containers"
+        super().__init__(*args, **kwargs)
+        self.job_flow_id = job_flow_id
+        self.step_id = step_id
+
+    async def emr_describe_step(self) -> Dict[str, Any]:
+        """
+        Make an API call with boto3 and get details about the cluster step.
+
+        For AWS API definition see here::
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr.html#EMR.Client.describe_step
+
+        :return: AWS EMR.Client.describe_step Api response Dict
+        """
+        async with await self.get_client_async() as client:
+            try:
+                response: Dict[str, Any] = await client.describe_step(
+                    ClusterId=self.job_flow_id, StepId=self.step_id
+                )
+                self.log.debug("Response from AwsBaseHookAsync: %s", response)
+                return response
+            except ClientError as error:
+                raise error
+
+    @staticmethod
+    def state_from_response(response: Dict[str, Any]) -> str:
+        """
+        Get state from response dictionary.
+        :param response: response from AWS API
+        :return: execution state of the cluster step
+        """
+        state: str = response["Step"]["Status"]["State"]
+        return state
+
+    @staticmethod
+    def failure_message_from_response(response: Dict[str, Any]) -> Optional[str]:
+        """
+        Get failure message from response dictionary.
+        :param response: response from AWS API
```

Comment:
```suggestion
        Get failure message from response dictionary.

        :param response: response from AWS API
```

Reply:
Sphinx requires a blank line to render properly

Reply:
Updated the docstring e9eb74b1fe0f7b7d015eeb34cc2acde939371b7e